pom.xml

```diff
@@ -1313,7 +1313,25 @@
                 </goals>
                 <phase>process-resources</phase>
                 <configuration>
-                  <flattenMode>oss</flattenMode>
+                  <!-- flattenMode:oss does not keep dependecnyMangement or will promote all transitive dependencies to direct ones, changing resolution order -->
+                  <pomElements>
+                    <dependencyManagement>interpolate</dependencyManagement>
```

Comment:
So what does the result actually look like in the original example? Does it effectively inline all deps from the imported BOM? We do not wish for a flattened POM to retain any reference to another POM file—it should be entirely self-contained.

Reply:
given we have an extension we **could** write our own [`DefaultInheritanceAssembler;`](https://github.com/mojohaus/flatten-maven-plugin/blob/master/src/main/java/org/codehaus/mojo/flatten/DirectDependenciesInheritanceAssembler.java) that is hpi aware, however I would consider that more risky than just including the `dependencyManagement` section - which at the end of the day we have been doing for years for plugins released with `m-r-p`

Reply:
https://repo.jenkins-ci.org/incrementals/io/jenkins/plugins/okhttp-api/4.10.0-126.v8a_8a_2605a_a_e7/  (interpolate)
https://repo.jenkins-ci.org/incrementals/io/jenkins/plugins/okhttp-api/4.10.0-126.v01d52e6d79e4/  (expand)

whilst expand is a lot and a lot of unused dependencies - as they are not transitive dependencies it will not matter if you favour completeness over simplicity.

Reply:
discussed directly with @jglick  as the intention of the flatten plugin here is to have a self contained pom - this will switch to `expand` which will also have the benefit of picking up any `<dependencyManagement>` configured in the parent(s).